src/overlays/actors/ovl_Bg_Jya_Megami/z_bg_jya_megami.c

```diff
@@ -21,31 +37,325 @@ const ActorInit Bg_Jya_Megami_InitVars = {
+        if (Math_Rand_ZeroOne() < arg2) {
+            Math_Vec3f_Sum(&this->dyna.actor.posRot.pos, &sPiecesInit[i].unk_00, &sp50);
+            sp50.z += 15.0f;
+            func_8089A1DC(globalContext, &sp50, &D_8089B184, 1, 0);
+        }
+    }
+}
+
+void BgJyaMegami_Init(Actor* thisx, GlobalContext* globalContext) {
+    BgJyaMegami* this = THIS;
+
+    BgJyaMegami_SetupDynapoly(this, globalContext, &D_06005C4C, 0);
+    BgJyaMegami_SetupCollider(this, globalContext);
+    if (Flags_GetSwitch(globalContext, this->dyna.actor.params & 0x3F)) {
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+        Actor_SetHeight(&this->dyna.actor, -50.0f);
+        BgJyaMegami_SetupDetectLight(this);
+    }
+}
+
+void BgJyaMegami_Destroy(Actor* thisx, GlobalContext* globalContext) {
+    BgJyaMegami* this = THIS;
+
+    DynaPolyInfo_Free(globalContext, &globalContext->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyJntSph(globalContext, &this->collider);
+}
+
+void BgJyaMegami_SetupDetectLight(BgJyaMegami* this) {
+    this->actionFunc = BgJyaMegami_DetectLight;
+    this->lightTimer = 0;
+    this->crumbleIndex = 0;
+}
+
+void BgJyaMegami_DetectLight(BgJyaMegami* this, GlobalContext* globalContext) {
+    if (this->collider.base.acFlags & 0x2) {
+        this->lightTimer++;
+        this->collider.base.acFlags &= ~0x2;
+        if (globalContext->gameplayFrames % 4 == 0) {
+            func_8089A41C(this, globalContext, (this->crumbleIndex * 0.04f) + 0.05f);
+        }
+        func_8002F974(&this->dyna.actor, NA_SE_EV_FACE_CRUMBLE_SLOW - SFX_FLAG);
+    } else if (this->lightTimer > 0) {
+        this->lightTimer--;
+    }
+    if (this->lightTimer > 40) {
+        Flags_SetSwitch(globalContext, this->dyna.actor.params & 0x3F);
+        BgJyaMegami_SetupExplosion(this);
+        Audio_PlaySoundAtPosition(globalContext, &this->dyna.actor.posRot.pos, 100, NA_SE_EV_FACE_EXPLOSION);
+        func_800800F8(globalContext, 0xD70, -0x63, &this->dyna.actor, 0);
+    } else {
+        if (this->lightTimer < 8) {
+            this->crumbleIndex = 0;
+        } else if (this->lightTimer < 16) {
+            this->crumbleIndex = 1;
+        } else if (this->lightTimer < 24) {
+            this->crumbleIndex = 2;
+        } else if (this->lightTimer < 32) {
+            this->crumbleIndex = 3;
+        } else {
+            this->crumbleIndex = 4;
+        }
+        CollisionCheck_SetAC(globalContext, &globalContext->colChkCtx, &this->collider.base);
+    }
+}
+
+void BgJyaMegami_SetupExplosion(BgJyaMegami* this) {
+    u32 i;
+
+    this->actionFunc = BgJyaMegami_Explosion;
+    for (i = 0; i < BGJYAMEGAMI_NUMPIECES; ++i) {
+        Math_Vec3f_Copy(&this->pieces[i].pos, &this->dyna.actor.posRot.pos);
+        this->pieces[i].vel.x = sPiecesInit[i].velX;
+    }
+    this->explosionTimer = 0;
+}
+
+void BgJyaMegami_Explosion(BgJyaMegami* this, GlobalContext* globalContext) {
```

Comment:
"Explosion" isn't a proper action name (fyi we call actions with verbs that correspond to the action of the actor itself, so things like "Fly", "Run", "Wait", etc...). "Explode" seems already better but it might not be accurate to what the actor is actually doing, so be mindful about that.
```suggestion
void BgJyaMegami_Explode(BgJyaMegami* this, GlobalContext* globalContext) {
```